### DISCUSSION TITLE
Simple bootstrapper with prerequisites and main MSI

### DISCUSSION THREAD

**Discussion by FredAtIvory** [wixtoolset/issues#8703] [Questions]
I have a C# WPF application with a WiX 5 installer that works great. I'm moving the application to .NET 8 and want to automatically install two prerequisites: (1) .NET 8 Desktop Runtime x86 and (2) WebView2. To do this, I created a Bundle using bal:WixInternalUIBootstrapperApplication:

```
<Wix
    xmlns="http://wixtoolset.org/schemas/v4/wxs"
    xmlns:bal="http://wixtoolset.org/schemas/v4/wxs/bal"
    xmlns:netfx="http://wixtoolset.org/schemas/v4/wxs/netfx">
    <Bundle
        Name="abc"
        Version="1.0.0.0"
        Manufacturer="xyz"
        UpgradeCode="beb84d01-360d-4ca9-88cc-ec14183a92bc">
        <BootstrapperApplication>
            <bal:WixInternalUIBootstrapperApplication/>
        </BootstrapperApplication>
        <Chain>
            <PackageGroupRef Id="DotNetDesktopRuntimePrereq" />
            <PackageGroupRef Id="WebView2Prereq" />
            <MsiPackage
                SourceFile="..\Setup\bin\x86\release\Setup.msi"
                Vital="yes"
                Compressed="yes"
                Permanent="no"
                Visible="yes"/>
        </Chain>
    </Bundle>
</Wix>
```

This is working in the sense that it installs the prerequisites as needed and shows the UI for the main application (Setup.msi), but I end up with an ARP entry for the bundle and the BA seems to be designed as a package manager which is overkill for what I need.

What I want is a simple wrapper/bootstrapper which installs the prerequisites and then runs the MSI. I want to end up with the prerequisites installed and a single ARP entry for my application which acts the same as if I had run the MSI directly.

What's the best way to achieve that? Should I use Burn or write something custom?

Thanks

**Comment by wmanning** [comment]
For wix3 short answer is no.
Long answer: https://robmensching.com/blog/posts/2012/6/25/b-is-for-bundle-and-thats-good-enough-for-me/

I think the same is true for wix v4 and 5.
